scheds/rust/scx_rustland/src/bpf/main.bpf.c

```diff
@@ -252,20 +252,17 @@ static void dispatch_local(struct task_struct *p, u64 enq_flags)
  */
 static void dispatch_on_cpu(struct task_struct *p, s32 cpu, u64 enq_flags)
 {
+	/*
+	 * If it's not possible to dispatch on the selected CPU, re-use the
+	 * previously used one.
+	 */
+	if (!bpf_cpumask_test_cpu(cpu, p->cpus_ptr))
+		cpu = scx_bpf_task_cpu(p);
```

Comment:
I think this should work fine but can you please verify that changing CPU affinity for a constantly running task which is bound to one CPU works? ie. Have one thread which busy loops, bind it to one cpu using `taskset`, and then change it to another CPU.

Reply:
Yep, it seems to work:
```
09:10 arighi@gpd$ yes >/dev/null &
[1] 1797739
6.7.0-3-generic ~
09:10 arighi@gpd$ taskset -pc 2 $!
pid 1797739's current affinity list: 0-7
pid 1797739's new affinity list: 2
6.7.0-3-generic ~
09:10 arighi@gpd$ cut -d' ' -f39 /proc/$!/stat
2
6.7.0-3-generic ~
09:11 arighi@gpd$ taskset -pc 5 $!
pid 1797739's current affinity list: 2
pid 1797739's new affinity list: 5
6.7.0-3-generic ~
09:11 arighi@gpd$ cut -d' ' -f39 /proc/$!/stat
5
```

Of course it changes cpu only if the task is doing something (i.e. a `sleep infinity` won't change cpu), but the same happens with the default scheduler, so we shouldn't change any behavior here.